haas/cli.py

```diff
@@ -215,17 +142,27 @@ def setup_http_client():
                            project_domain_id=os_project_domain_id)
         sess = session.Session(auth=auth)
         http_client = KeystoneHTTPClient(sess)
+        # For calls using the client library
+        C = Client(ep, http_client)
         return
     except (ImportError, KeyError):
         pass
     # Finally, fall back to no authentication:
     http_client = requests.Session()
+    C = Client(ep, http_client)
 
 
 class FailedAPICallException(Exception):
     pass
 
 
+def check_clientlib_response(fun):
+    try:
+        return fun()
+    except Exception as e:
+        sys.stderr.write('Error: %s\n' % e.message)
```

Comment:
By catching the exception you're causing the function to just return None, so on failure callers are going to plow ahead and probably raises some confusing exception. Probably the thing to do is call sys.exit instead of sys.stderr.write.